### PR TITLE
Handle MethodHandle signature-polymorphic native methods for JITServer

### PR DESCRIPTION
**Pull request by dmitry-ten**
PR #10617 modified a number of methods of ResolvedMethod classes.
This commit implements these changes for JITServer

Closes: #10941

### CODE REVIEW

**Comment by dmitry-ten** [comment]
@mpirvu fyi.
@liqunl could you also take a look?

File: runtime/compiler/env/j9method.cpp [hunk 1]
```diff
@@ -6689,15 +6699,12 @@ bool
 TR_ResolvedJ9Method::shouldCompileTimeResolveMethod(I_32 cpIndex)
    {
    I_32 realCPIndex = jitGetRealCPIndex(_fe->vmThread(), romClassPtr(), cpIndex);
```
Comment (liqunl) [inline runtime/compiler/env/j9method.cpp:6701]: This should be moved to `getMethodNameAndSignatureFromConstantPool`, and pass the cpIndex from bytecode to `getMethodNameAndSignatureFromConstantPool`

Comment (dmitry-ten) [reply runtime/compiler/env/j9method.cpp]: Done.

**Review by liqunl — APPROVED** [review]

**Review by mpirvu — APPROVED** [review]
LGTM

**Comment by mpirvu** [comment]
jenkins test sanity plinuxjit,xlinuxjit,zlinuxjit jdk11

**Comment by mpirvu** [comment]
One test failed on x86:
```
===============================================
Running test cmdLineTester_jvmtitests_debug_7 ...
===============================================
cmdLineTester_jvmtitests_debug_7 Start Time: Wed Oct 28 00:34:09 2020 Epoch Time (ms): 1603856049291
variation: Mode557
JVM_OPTIONS: -XX:+UseJITServer -XX:+UseCompressedOops -Xgcpolicy:balanced -Xdebug -Xrunjdwp:transport=dt_socket,address=8888,server=y,onthrow=no.pkg.foo,launch=echo -Xjit:count=0 
JITServer is currently a technology preview. Its use is not yet supported
JVMCDRT000E Unable to locate JIT stack map - aborting VM
JVMCDRT001E Method: java/lang/NullPointerException.<init>(Ljava/lang/String;)V (0000000000048128)
JVMCDRT002E Failing PC: 00007F747C9FE24E (offset 000000000000002E), metaData = 00007F745F6162B8
03:34:09.850 0x16700j9codertvm(j9ji.110    *   ** ASSERTION FAILED ** at /home/jenkins/workspace/Build_JDK11_x86-64_linux_jit_Personal/openj9/runtime/codert_vm/jswalk.c:532: ((0 ))
```
This is FSD mode and we have seen this kind of failure before. I will launch another round of testing on x86 to make sure we haven't made this problem more frequent that it was.

**Comment by mpirvu** [comment]
jenkins test sanity xlinuxjit jdk11

**Comment by mpirvu** [comment]
A different JVMTI test failed this time:
```
===============================================
Running test cmdLineTester_jvmtitests_8 ...
===============================================
cmdLineTester_jvmtitests_8 Start Time: Wed Oct 28 11:19:33 2020 Epoch Time (ms): 1603898373234
variation: Mode610
JVM_OPTIONS: -XX:+UseJITServer -Xcompressedrefs -Xjit -Xgcpolicy:gencon 
..........
Testing: emeng001
Test start time: 2020/10/28 11:20:56 Eastern Standard Time
Running command: "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_jit_Personal_testList_1/openjdkbinary/j2sdk-image/bin/java" -XX:+UseJITServer -Xcompressedrefs -Xjit -Xgcpolicy:gencon  -Xdump    -agentlib:jvmtitest=test:emeng001 -cp "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_jit_Personal_testList_1/openjdk-tests/TKG/../../jvmtest/functional/cmdLineTests/jvmtitests/jvmtitest.jar" com.ibm.jvmti.tests.util.TestRunner
Time spent starting: 4 milliseconds
Time spent executing: 309 milliseconds
Test result: FAILED
Output from test:
 [ERR] JITServer is currently a technology preview. Its use is not yet supported
 [ERR] Exception in thread "(unnamed thread)" java/lang/ExceptionInInitializerError
 [ERR] 	at java/lang/J9VMInternals.ensureError (java.base@9/J9VMInternals.java:185)
 [ERR] 	at java/lang/J9VMInternals.recordInitializationFailure (java.base@9/J9VMInternals.java:174)
 [ERR] java/lang/NullPointerException
 [ERR] 	at java/util/Objects.requireNonNull (java.base@9/Objects.java:4294967295)
 [ERR] 	at java/util/Arrays$ArrayList.<init> (java.base@9/Arrays.java:4354)
 [ERR] 	at java/util/Arrays.asList (java.base@9/Arrays.java:4341)
 [ERR] 	at java/lang/invoke/MethodTypeHelper.<clinit> (java.base@9/MethodTypeHelper.java:41)
>> Success condition was not found: [Return code: 0]
```

**Comment by mpirvu** [comment]
@dmitry-ten Could you please grind `cmdLineTester_jvmtitests_debug_7` locally? If it passes, I am going to merge this PR

**Comment by dmitry-ten** [comment]
Yeah, sure.

**Comment by dmitry-ten** [comment]
@mpirvu I actually ran into some build issues trying to test this locally today, so I just rebased to the latest master.

**Comment by dmitry-ten** [comment]
`cmdLineTester_jvmtitests_debug_7` passed when I grinded it locally.

**Comment by mpirvu** [comment]
jenkins test sanity xlinuxjit jdk11

**Comment by mpirvu** [comment]
This one failed again:
```
Testing: emeng001
Test start time: 2020/10/28 11:20:56 Eastern Standard Time
Running command: "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_jit_Personal_testList_1/openjdkbinary/j2sdk-image/bin/java" -XX:+UseJITServer -Xcompressedrefs -Xjit -Xgcpolicy:gencon  -Xdump    -agentlib:jvmtitest=test:emeng001 -cp "/home/jenkins/workspace/Test_openjdk11_j9_sanity.functional_x86-64_linux_jit_Personal_testList_1/openjdk-tests/TKG/../../jvmtest/functional/cmdLineTests/jvmtitests/jvmtitest.jar" com.ibm.jvmti.tests.util.TestRunner
Time spent starting: 4 milliseconds
Time spent executing: 309 milliseconds
Test result: FAILED
Output from test:
 [ERR] JITServer is currently a technology preview. Its use is not yet supported
 [ERR] Exception in thread "(unnamed thread)" java/lang/ExceptionInInitializerError
 [ERR] 	at java/lang/J9VMInternals.ensureError (java.base@9/J9VMInternals.java:185)
 [ERR] 	at java/lang/J9VMInternals.recordInitializationFailure (java.base@9/J9VMInternals.java:174)
 [ERR] java/lang/NullPointerException
 [ERR] 	at java/util/Objects.requireNonNull (java.base@9/Objects.java:4294967295)
 [ERR] 	at java/util/Arrays$ArrayList.<init> (java.base@9/Arrays.java:4354)
 [ERR] 	at java/util/Arrays.asList (java.base@9/Arrays.java:4341)
 [ERR] 	at java/lang/invoke/MethodTypeHelper.<clinit> (java.base@9/MethodTypeHelper.java:41)
>> Success condition was not found: [Return code: 0]
```

**Comment by mpirvu** [comment]
That `emeng001` passes for me on my local tests with code from this PR.
jenkins test sanity xlinuxjit jdk11

**Comment by mpirvu** [comment]
jenkins test sanity xlinuxjit jdk11

**Comment by mpirvu** [comment]
Tests passed, hence merging.

**Pull request merged**